### PR TITLE
Add initial API for NicDevice CRD and scaffolding for a config daemon reconciling it

### PR DESCRIPTION
**Pull request by almaslennikov**

### CODE REVIEW

File: api/v1alpha1/nicdevice_types.go [hunk 1]
```diff
@@ -20,22 +20,37 @@ import (
...
 
 // NicDeviceSpec defines the desired state of NicDevice
 type NicDeviceSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	Configuration *NicDeviceConfigurationSpec `json:"configuration,omitempty"`
+}
+
+type NicDevicePortSpec struct {
+	Pci              string `json:"pci"`
```
Comment (adrianchiris) [inline api/v1alpha1/nicdevice_types.go:34]: do we want to name it PCI or leave as is ?

File: api/v1alpha1/nicdevice_types.go [hunk 2]
```diff
@@ -20,22 +20,37 @@ import (
...
 
 // NicDeviceStatus defines the observed state of NicDevice
 type NicDeviceStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	Node            string                   `json:"node"`
+	Type            string                   `json:"type"`
+	SerialNumber    string                   `json:"serialNumber"`
+	PartNumber      string                   `json:"partNumber"`
+	Psid            string                   `json:"psid"`
```
Comment (adrianchiris) [inline api/v1alpha1/nicdevice_types.go:50]: These are initials of Product Serial Identification. do we want to name it PSID ? or leave as is ??

File: api/v1alpha1/nicdevice_types.go [hunk 3]
```diff
@@ -20,22 +20,37 @@ import (
...
+	// TODO add NicConfigurationTemplateSpec once #2 is merged
+}
 
 // NicDeviceSpec defines the desired state of NicDevice
 type NicDeviceSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	Configuration *NicDeviceConfigurationSpec `json:"configuration,omitempty"`
+}
+
```
Comment (adrianchiris) [inline api/v1alpha1/nicdevice_types.go:32]: please add docstring to everything

File: api/v1alpha1/nicdevice_types.go (hunk 1 above)
Comment (almaslennikov) [reply api/v1alpha1/nicdevice_types.go]: I don't have a preference, let's capitalize it

File: api/v1alpha1/nicdevice_types.go (hunk 2 above)
Comment (almaslennikov) [reply api/v1alpha1/nicdevice_types.go]: I don't have a preference, let's capitalize it

File: api/v1alpha1/nicdevice_types.go (hunk 3 above)
Comment (almaslennikov) [reply api/v1alpha1/nicdevice_types.go:41]: Thanks, added

File: api/v1alpha1/nicdevice_types.go [hunk 4]
```diff
@@ -20,22 +20,59 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+type NicDeviceConfigurationSpec struct {
```
Comment (adrianchiris) [inline api/v1alpha1/nicdevice_types.go:23]: nit missing struct docstring.

**Review by adrianchiris — APPROVED** [review]
left one nit, feel free to self merge

**Pull request merged**